lighthouse-cli/bin.js

```diff
@@ -94,34 +95,35 @@ if (cliFlags.extraHeaders) {
 /**
  * @return {Promise<LH.RunnerResult|void>}
  */
-function run() {
-  return Promise.resolve()
-    .then(_ => {
-      // By default, cliFlags.enableErrorReporting is undefined so the user is
-      // prompted. This can be overriden with an explicit flag or by the cached
-      // answer returned by askPermission().
-      if (typeof cliFlags.enableErrorReporting === 'undefined') {
-        return askPermission().then(answer => {
-          cliFlags.enableErrorReporting = answer;
-        });
-      }
-    })
-    .then(_ => {
-      Sentry.init({
-        url,
-        flags: cliFlags,
-        environmentData: {
-          name: 'redacted', // prevent sentry from using hostname
-          environment: isDev() ? 'development' : 'production',
-          release: pkg.version,
-          tags: {
-            channel: 'cli',
-          },
-        },
-      });
+async function run() {
+  if (cliFlags.printConfig) {
+    const config = generateConfig(cliFlags, configJson);
+    process.stdout.write(config.print());
+    return;
+  }
```

Comment:
this needed to be in the full flow of the cli since it turns out `process.stdout.write` (and by extension `console.log`) isn't flushed or waited on by `process.exit()`, so the exit can come mid write and the JSON output will be truncated. Since `index.js` just calls runs this module and calls `run()`, it needed some way to prevent getting to `run()` without calling `process.exit()`, so...here we are.

(and might as well switch to async/await while in here)